subsys/net/ip/net_pkt.c

```diff
@@ -888,6 +888,23 @@ void net_pkt_get_info(struct k_mem_slab **pkts,
 	}
 }
 
+struct net_pkt *net_pkt_pullup(struct net_pkt *pkt, int len)
+{
+	net_pkt_iter_init(pkt);
```

Comment:
That means you expect the contiguous space you are looking for being always at the beginning of the buffer. That's probably not what you want.

Just remove that, let the user place the iterator where it's supposed to  be (ip hdr, or udp one etc...) and call this function to verify contiguity (thus verify that pkt->iter.pos is fine and net_pkt_tod() will be giving you back the cast you want).

Reply:
Indeed, good point!